recipes/mcap/all/conanfile.py

```diff
@@ -0,0 +1,44 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class McapConan(ConanFile):
+    name = "mcap"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/foxglove/mcap"
+    description = "A C++ implementation of the MCAP file format"
+    license = "Apache-2.0"
+    topics = ("mcap", "serialization", "deserialization", "recording")
+
+    settings = ("os", "compiler", "build_type", "arch")
+    requires = ("fmt/8.1.1", "lz4/1.9.3", "zstd/1.5.2")
+    generators = ("cmake", "cmake_find_package")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _source_package_path(self):
+        return os.path.join(self._source_subfolder, "cpp", "mcap")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        if self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) <= 11:
+            raise ConanInvalidConfiguration("Compiler version is not supported, c++17 support is required")
+        if (self.settings.compiler == "gcc" or self.settings.compiler == "clang") and tools.Version(self.settings.compiler.version) <= 8:
+            raise ConanInvalidConfiguration("Compiler version is not supported, c++17 support is required")
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) <= "16.8":
```

Comment:
don't Visual Studio version like that, it raises for Visual Studio 2019.

Reply:
I got this from https://github.com/conan-io/conan/issues/8002, which is linked from the readme: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-is-a-toolscheck_min_cppstd-call-not-enough, via this conversation on another package I worked on: https://github.com/conan-io/conan-center-index/pull/8662#discussion_r782533406  cc @prince-chrismc